document_splitter/src/get_optimal_k.py

```diff
@@ -0,0 +1,113 @@
+"""
+This script contains the functions to get keyword representations and elbow point
+"""
+from typing import List, Dict
+import logging
+import math
+import pandas as pd
+from PIL import Image
+from sklearn.cluster import KMeans
+from keybert import KeyBERT
+from sentence_transformers import SentenceTransformer
+
+KEYWORD_EXTRACTION_MODEL = KeyBERT()
+log = logging.getLogger("Log ")
+
+
+def get_keywords(text: str) -> str:
+    """
+    Function to get keyword representation from full text
+    Args:
+        text (str): the text from which keywords need to be extracted
+    Returns:
+        keyword_representation (str): keywords extracted
+    """
+    keywords_extracted = KEYWORD_EXTRACTION_MODEL.extract_keywords(text, top_n=50)
+    keywords_list = list(set([(word[0]).lower() for word in keywords_extracted]))
+    keyword_representation = " ".join(keywords_list)
+    log.info(f"Keyword representation is - {keyword_representation} \n")
+
+    return keyword_representation
+
+
+def get_slope(x_list: List, y_list: List) -> Dict:
+    """
+    Function to get slope fo respective points in two lists
+    Args:
+        x_list (List): list of k values
+        y_list (List): list of qcss scores
+    Returns:
+        slope_dict (Dict): Slope values to get elbow point
+    """
+    slope_dict = {}
+
+    for i in range(0, len(x_list) - 1):
```

Comment:
You could write this as a dict-comprehension, since non of these variables are needed beyond the context of creating a dictionary. Also, since checking for key availability is not necessary, you could use `ordereddict`

Reply:
I did it without comprehension as I felt this way improved visibility of the logic being used. But thank you I will keep this suggestion in mind :)